### PR TITLE
content/productivity: Adding a chapter on productivity tools

### PR DESCRIPTION
**Pull request by simo-tuomisto**
This PR adds a new chapter on productivity tools such as linters and code formatters.

### CODE REVIEW

**Review by bast — APPROVED** [review]
This is great!

One thing that can be useful in the context of Notebooks (where most of this course takes place), adding these dependencies adds you a button in notebooks to autoformat code:
```
jupyterlab-code-formatter
black
isort
```

**Comment by simo-tuomisto** [comment]
> This is great!
> 
> One thing that can be useful in the context of Notebooks (where most of this course takes place), adding these dependencies adds you a button in notebooks to autoformat code:
> 
> ```
> jupyterlab-code-formatter
> black
> isort
> ```

Unfortunately `jupyterlab-code-formatter` does not seem to be available from `defaults`-channel. It is available from conda-forge as `jupyterlab_code_formatter`. Adding it as a pip package might be bit risky.

**Comment by simo-tuomisto** [comment]
I'll add mention of isort.

**Comment by simo-tuomisto** [comment]
> I'll add mention of isort.

Added mentions of both tools.

File: content/productivity.rst [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+   import matplotlib.pyplot as plt
+
+   x = np.linspace(0, np.pi, 100))
+   y = np.sin(x)
+
+   plt.plot(x, y)
+
+   plt.show()
+
+.. code-block:: sh
```
Comment (rkdarst) [inline content/productivity.rst:58]:
```suggestion
.. code-block:: console
```

File: content/productivity.rst [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+on line 4. We also get an message code E0001 (syntax-error). We can find
+description for the message from
+`Pylint's messages list <https://pylint.readthedocs.io/en/stable/user_guide/messages/messages_overview.html#error-category>`__
+and look at the
+`specific error page <https://pylint.readthedocs.io/en/stable/user_guide/messages/error/syntax-error.html>`__ to see an example that describes the error.
+
+After fixing the problem with the bracket and running ``pylint`` again we
+get more errors:
+
+.. code-block:: sh
```
Comment (rkdarst) [inline content/productivity.rst:75]:
```suggestion
.. code-block:: console
```

File: content/productivity.rst [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+
+From these messages we can deduce that the main problem is that the import
+statement does not use ``import numpy as np`` and thus ``np`` is undefined.
+
+After changing the import stamement, the code works correctly and running
+``pylint lint_example.py`` will only warn about the missing docstring.
+You can also notice that the changes have increased the rating and
+Pylint will show the improvement since last run.
+
+.. code-block:: sh
```
Comment (rkdarst) [inline content/productivity.rst:111]:
```suggestion
.. code-block:: console
```

File: content/productivity.rst [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+   for number  in range(1,8):
+
+       n = 10** number
+
+       print(f'Estimate for PI with {n:8d} dart throws: {PI_estimate( n )}')
+
+
+Running flake8 to check for style problems we get the following output:
+
+.. code-block:: bash
```
Comment (rkdarst) [inline content/productivity.rst:307]:
```suggestion
.. code-block:: console
```

File: content/productivity.rst [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+   `a bit different to PEP 8 <https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html>`__
+   and one can definitely argue that it
+   `does not handle mathematical expressions in the optimal way <https://github.com/psf/black/issues/148>`__.
+
+   However, one can turn formatting off for math heavy sections with ``# fmt: on``-
+   and ``# fmt: off``-comments. Alternatively, you can use formatter such as
+   `yapf <https://github.com/google/yapf>`__, which supports formatting based on
+   arithmetic precedence:
+
+   .. code-block:: bash
```
Comment (rkdarst) [inline content/productivity.rst:411]:
```suggestion
   .. code-block:: console
```

**Review by rkdarst — APPROVED** [review]

**Pull request merged**